### PR TITLE
Update WordPressKit and WordPressAuthenticator

### PR DESCRIPTION
**Pull request by crazytonyli**
The latest WordPressKit introduces a breaking change which requires code changes in the apps and WordPressAuthenticator.

This PR updates WordPressKit to the latest version and uses a WordPressAuthenticator commit (part of https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/829) that adopts to the WordPressKit breaking changes. We can merge this PR after https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/829 is merged.

## Regression Notes
None

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist: N/A

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">PR requires at least one label.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by kean** [comment]
Thanks, @crazytonyli !

Are there any specific test steps? I ran a smoke test and it LGTM.

**Review by kean — APPROVED** [review]

File: Podfile [hunk 1]
```diff
@@ -50,9 +50,9 @@ def wordpress_ui
 end
 
 def wordpress_kit
-  pod 'WordPressKit', '~> 11.0'
+  pod 'WordPressKit', '~> 12.0'
   # pod 'WordPressKit', git: 'https://github.com/wordpress-mobile/WordPressKit-iOS.git', commit: ''
-  # pod 'WordPressKit', git: 'https://github.com/wordpress-mobile/WordPressKit-iOS.git', branch: 'trunk'
+  # pod 'WordPressKit', git: 'https://github.com/wordpress-mobile/WordPressKit-iOS.git', branch: 'wordpress-com-rest-api-error-refactor'
```
Comment (kean) [inline Podfile]: Please this line and line 144.

Comment (crazytonyli) [reply Podfile]: Can you clarify what needs to be done regarding line 144? I'll merge this PR for now to unblock others. I'm happy to address any further issues.

**Pull request merged**